### ISSUE TITLE
cloudflare_teams_rule having block_page_enabled to false is not idempotent

### ISSUE THREAD

**Issue by DamienDelporte** [cloudflare/terraform-provider-cloudflare#2923]
Cloudflare Entreprise ticket: [3019496](https://support.cloudflare.com/hc/en-us/requests/3019496)

### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

```
$ terraform version
Terraform v1.6.0
on linux_amd64
+ provider registry.terraform.io/cloudflare/cloudflare v4.18.0
```

### Affected resource(s)

cloudflare_teams_rule

### Terraform configuration files

```hcl
provider "cloudflare" {
}

terraform {
  required_providers {
    cloudflare = {
      source  = "cloudflare/cloudflare"
      version = "= 4.18.0"
    }
  }
}

resource "cloudflare_teams_rule" "tf_http_allow" {
  account_id  = "XXX"
  name        = "test"
  description = "test"
  enabled     = true
  precedence  = 50
  action      = "allow"
  filters     = ["http"]
  traffic     = "http.request.uri == \"https://www.example.com/not-malicious\""
  rule_settings {
    block_page_enabled = false
  }
}
```

### Link to debug output

Not needed.

### Expected output

No changes when applying a second time (without anything new changed).

### Actual output
```
$ terraform apply
cloudflare_teams_rule.tf_http_allow: Refreshing state... [id=XXX]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # cloudflare_teams_rule.tf_http_allow will be updated in-place
  ~ resource "cloudflare_teams_rule" "tf_http_allow" {
        id          = "XXX"
        name        = "test"
        # (8 unchanged attributes hidden)

      + rule_settings {
          + block_page_enabled = false
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: 
```

### Steps to reproduce

1. terraform apply (to create)
2. terraform apply/plan (and see the change...)

**Comment by DamienDelporte** [comment]
Cloudflare Entreprise ticket: [3019496](https://support.cloudflare.com/hc/en-us/requests/3019496)

**Comment by DamienDelporte** [comment]
Still not fixed in 4.22 even if https://github.com/cloudflare/terraform-provider-cloudflare/pull/3010 should have.

```
$ terraform apply
cloudflare_teams_rule.tf_http_allow: Refreshing state... [id=xxx]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # cloudflare_teams_rule.tf_http_allow will be updated in-place
  ~ resource "cloudflare_teams_rule" "tf_http_allow" {
        id          = "xxx"
        name        = "test"
        # (8 unchanged attributes hidden)

      + rule_settings {
          + block_page_enabled = false
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: no-it-is-not-fixed

Apply cancelled.
```
